skops/hf_hub/_hf_hub.py

```diff
@@ -207,3 +207,97 @@ def push(
         revision=None,
         create_pr=False,
     )
+
+
+def get_config(path: Union[str, Path]):
+    """Returns the configuration of a project.
+
+    Parameters
+    ----------
+    path: str
+        The path to the director holding the project and its ``config.json``
+        configuration file.
+
+    Returns
+    -------
+    config: dict
+        A dictionary which holds the configs of the project.
+    """
+    with open(Path(path) / "config.json", "r") as f:
+        config = json.load(f)
+    return config
+
+
+def get_requirements(path: Union[str, Path]):
+    """Returns the requirements of a project.
+
+    Parameters
+    ----------
+    path: str
+        The path to the director holding the project and its ``config.json``
+        configuration file.
+
+    Returns
+    -------
+    requirements: list of str
+        The list of requirements which can be passed to the package manager to
+        be installed.
+    """
+    config = get_config(path)
+    return config.get("sklearn", dict()).get("environment", dict())
+
+
+def download(
+    *,
+    repo_id: str,
+    dst: Union[str, Path],
+    revision: str = None,
+    token: str = None,
+    keep_cache: bool = True,
+    **kwargs,
+):
+    """Download a repository into a directory.
+
+    The directory needs to be an empty or a non-existing one.
+
+    Parameters
+    ----------
+    repo_id: str
+        The ID of the Hugging Face Hub repository in the form of
+        ``OWNER/REPO_NAME``.
+
+    dst: str, or Path
+        The directory to which the files are downloaded.
+
+    revision: str
+        The revision of the project to download. This can be a git tag, branch,
+        or a git commit hash.
+
+    token: str, optional
+        The token to be used to download the files. Only required if the
+        repository is private.
+
+    keep_cache: bool, default=True
```

Comment:
Maybe mention it's a git-based layout and that it will only redownload if there are new commits in remote repo

Reply:
Perfect! :fire: